public/js/helpers/chart_helper.js

```diff
@@ -53,3 +53,101 @@ export function padPoints (pts, binSize, sustain) {
   pts.unshift(front)
   pts.push(back)
 }
+
+var zoomMap = {
+  all: 0,
+  year: 3.154e+10,
+  month: 2.628e+9,
+  week: 6.048e+8,
+  day: 8.64e+7
+}
+
+export class Zoom {
+  static object (start, end) {
+    return {
+      start: start,
+      end: end
+    }
+  }
+
+  static map (key) {
+    return zoomMap[key]
+  }
+
+  static encode (start, end) {
+    if (!end) {
+      start = this.tryDecode(start)
+      end = start.end
+      start = start.start
+    }
+    return parseInt(start / 1000).toString(36) + '-' + parseInt(end / 1000).toString(36)
+  }
+
+  static decode (encoded, lims) {
+    // if lims are provided, encoded can be a zoomMap key
+    encoded = this.tryDecode(encoded)
+    lims = this.tryDecode(lims)
```

Comment:
What does `lims` mean? I don't see this 2nd arg being passed anywhere and it's not clear to me what it's supposed to represent based in the name of the variable.

Reply:
Current x-axis limits. The second parameter is used in `Zoom.validate` and allows the keywords `day`, `week`, `month`, `year`, and `all`  to be used to set the range. Those values map the the last possible window of that size in the available chart range.  

Should I rename it or document it better?

Reply:
Good for now IMO until we deploy and get more feedback